### ISSUE TITLE
Instability in performance (Multivariate Regression)

### ISSUE THREAD

**Issue by CoteDave** [stanfordmlgroup/ngboost#106]
Hi, 

I tried the NGBRegressor on the Boston dataset, and it perform very well, like the demo example, It offer similar / best results thant XGBoost or RandomForest.

When I try NGBRegressor on an other very simple regression dataset from kaggle (https://www.kaggle.com/mirichoi0218/insurance) it perform very very very bad compared to Random Forest or XGBoost or .... LinearRegression !!

Is there a reason of this dramatic change in performance ? Can we adjust the hyper-parameter to get good results as XGBoost / RandomForest ? What is going on ? Is it because outliers have a bigger impact on NGBRegressor than XGBoost or RandomForest ?

here is the code:
dataset = pd.read_csv('insurance_clean.csv', sep = ';')
X = dataset[dataset.columns.difference(['charges'])].values
feature_names = list(dataset.columns.difference(['charges']))
y = dataset[['charges']].values.ravel()

SEED = 2019
 
# Train/test split
X_train, X_test, y_train, y_test = train_test_split(X, y,
                                                    test_size=0.2,
                                                    random_state=SEED)
print('The shape of training dataset: {}'.format(X_train.shape[0]))
print('The shape of testing dataset: {}'.format(X_test.shape[0]))
# Fit and predict
np.random.seed(SEED)
ngb = NGBRegressor(n_estimators=400,
                   Base=default_tree_learner, Dist=Normal, Score=MLE).fit(X_train, y_train)
y_pred = ngb.predict(X_test)
print('NGBoost: R2 score on testing data: {:.2f}%'.format(100 * r2_score(y_test, y_pred)))

Thanks !

**Comment by CoteDave** [comment]
Got better results with Exponential and LogNormal distributions... would be nice to have a 'Auto' option for the Dist parameter, that auto-select the best distribution.

Thanks!

**Comment by alejandroschuler** [comment]
@CoteDave there could be a number of reasons for this. 

Although, if you are only interested in prediction RMSE and not in probabilistic prediction (i.e. prediction intervals) you should not use ngboost. XGBoost will almost always be strictly better for RMSE. Although NGBoost can be used for standard point prediction and it seems to perform only slightly worse than dedicated approaches like XGBoost, it is designed to solve a different problem altogether.
